doc/docs-revamp/dev-onboarding-quick-setup.md

```diff
@@ -0,0 +1,375 @@
+- [Nix setup guide](https://github.com/input-output-hk/iogx/blob/main/doc/nix-setup-guide.md): instructions for installing and configuring Nix to work with projects at IOG.
+
+#### Haskell tool stack ("Stack")
+
+Stack is a cross-platform program for developing Haskell projects. It handles dependency management, building, and testing your Haskell code. 
+
+To install the Stack: 
+1. Visit the [official Haskell Stack website](https://docs.haskellstack.org/). 
+2. Follow the installation instructions for your operating system (Linux, macOS, or Windows).
+3. Verify your installation by running `stack --version` in your terminal.
+
+#### GHC (Glasgow Haskell Compiler)
+
+- GHC is the main compiler for Haskell with many extensions and optimizations. 
+- To install GHC, run `stack setup`. Stack will automatically install the appropriate version of GHC for your project. 
+
+#### Cabal
+
+Cabal is a system for building and packaging Haskell libraries and programs. It works well with Stack and is used to manage project dependencies and build configurations. Cabal is installed automatically when you install Stack. 
+
+To verify your Cabal installation, run `cabal –version`. 
+
+### Git
+
+Make sure you have Git installed and configured. See [https://git-scm.com/](https://git-scm.com/). 
+
+### IDE
+
+An IDE provides a comprehensive environment for writing, testing, and debugging your Haskell code. Visual Studio Code (VS Code) is a widely used code editor that supports the Haskell extension. See [https://code.visualstudio.com/](https://code.visualstudio.com/). 
+
+## Testing your environment setup
+
+To verify that your environment is set up correctly and functioning, create and run a simple Haskell program. 
+
+The following simple program fetches and displays the current system time and date. This example involves importing and using libraries to handle dates and times. 
+
+### To create your test program
+
+1. Create a new file named `Main.hs`. 
+2. Add the following code to the file: 
+
+```haskell
+import Data.Time
+main :: IO ()
+main = do
+    currentTime <- getCurrentTime
+    putStrLn $ "Current system time is: " ++ show currentTime
+```
+
+3. Open a terminal in VS Code. 
+4. Navigate to the directory containing `Main.hs`. 
+5. Run the command `stack runghc Main.hs`.
+6. Verify that the output is correct. 
+
+## Core concepts of Plutus smart contracts
+Before we start building our vesting smart contract, let's go over the following core concepts that underpin Plutus smart contracts.
+
+### Datum
+In a vesting contract, the datum represents the state and conditions of the vesting agreement. It includes information such as the beneficiary's public key hash and the deadline for the vesting period. The datum is attached to the UTxO that holds the vested funds and is used by the contract's validator script to enforce the vesting conditions. It ensures that the funds can only be released to the beneficiary after the specified deadline has passed.
```

Comment:
If we want to have a subsection explaining what Datum is, it's better to explain what it is in general, and then proceed to explain what it does in the specific context of a vesting contract.

Besides, if there's a section explaining core concepts, I think it should start with the EUTxO model.